### PR TITLE
Feat/zap automation

### PR DESCRIPTION
**Pull request by jusdino**
Add ZAP scan automation

### Notes
- This PR requires GitHub secrets to be established in CSG's repo to complete scan integration. See the new ZAP readme for details.
- The GitHub Action does not support customization of failure level, so we will need to work together to address/suppress each alert before we will see a successful scan result.

### Description List
- Leverage the [Automation Framework](https://www.zaproxy.org/docs/desktop/addons/automation-framework/) to create a scan plan
- Add a GitHub Action to scan on PRs against `main`
- Add a script to run the same scan locally (in docker)
- Create a VULNERABLE security profile that enables a more effective scan
- Add checks and tests to prevent VULNERABLE profile from being deployed to production

### Testing List
- Standard test suite
- Open a PR against `main` and review scan logs and report

Closes #
#227

### CODE REVIEW

File: backend/compact-connect/lambdas/cloudfront-csp/test/index.test.js [hunk 1]
```diff
@@ -209,27 +209,6 @@ describe(testFilename(__filename), () => {
 
             checkLambdaResult(environment, result);
         });
-        it('should successfully return the security headers for ia prod', async () => {
```
Comment (jlkravitz) [inline backend/compact-connect/lambdas/cloudfront-csp/test/index.test.js:212]: why remove these tests?

File: backend/compact-connect/lambdas/staff-user-pre-token/user_scopes.py [hunk 2]
```diff
@@ -8,7 +8,9 @@ class UserScopes(set):
     Custom Set that will populate itself based on the user's database record contents
     """
     def __init__(self, sub: str):
-        super().__init__()
+        # Some auth flows (like SRP) don't grant 'profile', so we'll make sure it's included by default
```
Comment (jlkravitz) [inline backend/compact-connect/lambdas/staff-user-pre-token/user_scopes.py:11]: What's SRP?

File: owasp-zap/data/bearer-token.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+// var ScriptVars = Java.type('org.zaproxy.zap.extension.script.ScriptVars');
+const System = Java.type('java.lang.System');
+
+const token = System.getenv('ZAP_AUTH_HEADER_VALUE');
+
+
+function sendingRequest(msg, initiator, helper) {
+    // add Authorization header to all request in scope except the authorization request itself
+    if (initiator !== HttpSender.AUTHENTICATION_INITIATOR && msg.isInScope()) {
+        // const token = ScriptVars.getGlobalVar('accessToken');
```
Comment (jlkravitz) [inline owasp-zap/data/bearer-token.js:17]: remove line?

File: owasp-zap/data/bearer-token.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Adapted from https://github.com/zaproxy/community-scripts/blob/main/httpsender/AddBearerTokenHeader.js
+ * This script adds a bearer token to all requests in scope except the authorization request itself
+ * The token is retrieved from the environment variable TOKEN, which can be acquired separately from ZAP.
+ */
+
+var HttpSender = Java.type('org.parosproxy.paros.network.HttpSender');
+// var ScriptVars = Java.type('org.zaproxy.zap.extension.script.ScriptVars');
```
Comment (jlkravitz) [inline owasp-zap/data/bearer-token.js:8]: remove commented code

File: owasp-zap/session/.gitkeep
Comment (jlkravitz) [inline owasp-zap/session/.gitkeep:1]: What's the session directory used for?

File: owasp-zap/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# OWASP Zed Attack Proxy
+
+[ZAP](https://www.zaproxy.org/) is a security pen testing tool, which this project uses for active scanning for vulnerabilities. To manually run a scan similar to the ones integrated into [our GitHub Actions](../.github/workflows), run `./manual-scan.sh` from a command line. For full details on running the scan locally, see [Manual run ](#manual-run). Below is a brief explanation of the pieces of this integration and what they are for.
+
+# Authenticator
+
+The [authenticator](./authenticator) directory is a simple NodeJS script that leverages [aws-amplify](https://www.npmjs.com/package/aws-amplify) to quickly log a test user in and acquire an token for authentication. This script requires that the Cognito UserPool Client be configured to enable the SRP authentication flow as part of a 'VULNERABLE' security profile, which can only be activated in pre-production environments. See [Set up](#set-up) for discussion of the profile.
```
Comment (jlkravitz) [inline owasp-zap/README.md:7]: I found SRP online...can we spell it out at least once in the documentation, e.g., "Secure Remote Password (SRP)" and maybe link to AWS docs.

**Review by jlkravitz — COMMENTED** [review]
several small questions/comments but looks great! thanks for digging into this. is there anything we need to do (e.g., changing the security profile on the test deployment) to get this up-and-running once merged?

File: owasp-zap/session/.gitkeep
Comment (jusdino) [reply owasp-zap/session/.gitkeep:1]: Oh, I guess we don't need that anymore. I was using it for persisting local ZAP session data.

File: backend/compact-connect/lambdas/cloudfront-csp/test/index.test.js (hunk 1 above)
Comment (jusdino) [reply backend/compact-connect/lambdas/cloudfront-csp/test/index.test.js:212]: This test is for an environment that doesn't exist, so I removed it.

**Comment by jusdino** [comment]
> several small questions/comments but looks great! thanks for digging into this. is there anything we need to do (e.g., changing the security profile on the test deployment) to get this up-and-running once merged?

To get this up and running, we'll need to create the GitHub Secrets mentioned in the PR description and to update the `test` environment config to use the new `VULNERABLE` security profile. We can team up and I'll walk you through it 😄

**Review by isabeleliassen — APPROVED** [review]

**Pull request merged**